metrics_utility/automation_controller_billing/extract/extractor_common.py

```diff
@@ -87,29 +87,40 @@ def process_tarballs(self, path, temp_dir):
     # for index, row in chunk.iterrows():
     #     print(row)
 
-    if os.path.exists(os.path.join(temp_dir, 'job_host_summary.csv')):
-        job_host_summary = pd.read_csv(os.path.join(temp_dir, 'job_host_summary.csv'))
-    else:
-        job_host_summary = pd.DataFrame([{}])
-
-    if os.path.exists(os.path.join(temp_dir, 'main_indirectmanagednodeaudit.csv')):
-        indirect_nodes = pd.read_csv(os.path.join(temp_dir, 'main_indirectmanagednodeaudit.csv'))
-    else:
-        indirect_nodes = pd.DataFrame([{}])
-
-    if os.path.exists(os.path.join(temp_dir, 'main_jobevent.csv')):
-        main_jobevent = pd.read_csv(os.path.join(temp_dir, 'main_jobevent.csv'))
-    else:
-        main_jobevent = pd.DataFrame([{}])
-
-    if os.path.exists(os.path.join(temp_dir, 'main_host.csv')):
-        main_host = pd.read_csv(os.path.join(temp_dir, 'main_host.csv'))
-    else:
-        main_host = pd.DataFrame([{}])
-
-
-    return {'main_jobevent': main_jobevent,
-            'main_host': main_host,
-            'job_host_summary': job_host_summary,
-            'indirect_nodes' : indirect_nodes,
-            'config': config}
+    needed_data = {'config': config}
+    if required_sheets(self,['ccsp_summary','managed_nodes_by_organizations','managed_nodes','usage_by_organizations']):
+        needed_data['job_host_summary'] = build_data_batch(temp_dir,'job_host_summary')
+    if required_sheets(self,['usage_by_collections','usage_by_roles','usage_by_modules']):
+        needed_data['main_jobevent'] = build_data_batch(temp_dir,'main_jobevent')
+    if required_sheets(self,['indirectly_managed_nodes']):
+        needed_data['indirect_nodes']= build_data_batch(temp_dir,'indirect_nodes')
+    if required_sheets(self,['inventory_scope']):
+        needed_data['main_host']= build_data_batch(temp_dir,'main_host')
+
+    return needed_data
+
+
+def build_data_batch(temp_dir, file_name):
+      """
+      Builds the report with only the necessary sheets.
+      """
+
+      if os.path.exists(os.path.join(temp_dir, f'{file_name}.csv')):
+          return pd.read_csv(os.path.join(temp_dir, f'{file_name}.csv'))
+      else:
+          return pd.DataFrame([{}])
+
+
+def required_sheets(self, sheets_required):
+    """
+    Checks for METRICS_UTILITY_OPTIONAL_CCSP_REPORT_SHEETS values and return those values.
+    If none found, give it the default CCSP report sheet options.
+    Returns a boolean so we know which sheets to provide in the report.
+
+    """
+    sheet_options = os.getenv('METRICS_UTILITY_OPTIONAL_CCSP_REPORT_SHEETS')
+    if sheet_options is None:
+        sheet_options = self.optional_report_sheets()
```

Comment:
one of these returns a comma-separated string, the other returns an array of strings

```suggestion
    sheet_options = self.optional_report_sheets()
```